pkg/thing/thing.go

```diff
@@ -248,8 +251,14 @@ type Builder interface {
 	// a sub-realm of root called "alfheim"; "/alfheim"
 	// a sub-realm of alfheim called "svartalfheim"; "/alfheim/svartalfheim"
 	InRealm(realm string) Builder
-	// AuthenticateWith the named authentication tree
-	AuthenticateWith(tree string) Builder
+	// WithTree sets the tree that the thing authenticates with
+	WithTree(tree string) Builder
+	// AsService registers the thing as a service. By default, a thing is registered as a device
+	AsService() Builder
```

Comment:
I know we discussed adding a `AsDevice()` function as well but it felt a bit redundant since it was the default state.